node/src/components/block_proposer.rs

```diff
@@ -253,6 +243,7 @@ impl BlockProposer {
             .filter(|&(hash, deploy)| {
                 self.is_deploy_valid(deploy, block_timestamp, &deploy_config, &past_deploys)
                     && !past_deploys.contains(hash)
+                    && !self.state.finalized_deploys.contains_key(hash)
```

Comment:
We really should be using that optimized collection that Fraser wrote in maidsafe. It is rather wasteful to hash an already unique hash.

Reply:
Which collection do you mean?

I could do that as a part of this PR if more people think it's a good idea. Otherwise, maybe let's ticket it as an improvement for the future?

Reply:
Yes, we should definitely ticket it :

Reply:
https://crates.io/crates/hash_hasher